src/mlpack/tests/fastmks_test.cpp

```diff
@@ -341,9 +341,9 @@ BOOST_AUTO_TEST_CASE(FastMKSModelLinearTest)
   FastMKSModel mNaive(FastMKSModel::LINEAR_KERNEL);
   FastMKSModel mSingle(FastMKSModel::LINEAR_KERNEL);
 
-  m.BuildModel(referenceData, lk, false, false, 2.0);
-  mNaive.BuildModel(referenceData, lk, false, true, 2.0);
-  mSingle.BuildModel(referenceData, lk, true, false, 2.0);
+  m.BuildModel(std::move(referenceData), lk, false, false, 2.0);
+  mNaive.BuildModel(std::move(referenceData), lk, false, true, 2.0);
+  mSingle.BuildModel(std::move(referenceData), lk, true, false, 2.0);
```

Comment:
When you call `std::move()`, the internal parts of the object are transferred to the `BuildModel()` method, and so after you call `std::move(referenceData)`, then `referenceData` is an empty object.  Therefore, in situations where you need to use `referenceData` more than once, don't use `std::move()`.  I think this will fix the failing tests.

Reply:
Thanks for the help. Updated :)